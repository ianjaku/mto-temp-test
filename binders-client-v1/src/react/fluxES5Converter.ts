/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
