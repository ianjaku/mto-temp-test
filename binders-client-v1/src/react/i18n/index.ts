import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
