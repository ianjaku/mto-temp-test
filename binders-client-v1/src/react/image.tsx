/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
