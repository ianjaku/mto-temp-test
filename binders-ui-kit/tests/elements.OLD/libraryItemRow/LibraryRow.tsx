import * as React from "react";
import LibraryRow, { ILibraryRowAction, ItemType } from "../../../src/elements/libraryItem/row";
import MuiThemeProvider from "@material-ui/core/styles/MuiThemeProvider";
import { create } from "react-test-renderer";
import { mount } from "enzyme";

const THUMBNAIL = {
    bgColor: "white",
    buildRenderUrl: () => "https://dummyimage.com/100x100",
    fitBehaviour: "fit",
    medium: "https://dummyimage.com/100x100",
};

const TITLE = "a document";
const WIDTH = "600px";

let clickedRow = false;

const actions: ILibraryRowAction[] = [
    {
        iconName: "edit",
        onClick: undefined,
        title: "action 1",
    },
    {
        iconName: "delete",
        onClick: undefined,
        title: "action 2",
    },
];

function wrapWithTheme(node) {
    return (
        <MuiThemeProvider>{node}</MuiThemeProvider>
    );
}

const createLibraryRow = () => wrapWithTheme(<LibraryRow title={TITLE} thumbnail={THUMBNAIL}/>);
const createLibraryRowFixedWidth = () => wrapWithTheme(<LibraryRow title={TITLE} thumbnail={THUMBNAIL} width={WIDTH}/>);
const createLibraryRowWithActions = () => wrapWithTheme(<LibraryRow title={TITLE} thumbnail={THUMBNAIL} libraryRowActions={actions}/>);
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const createLibraryFolderRow = () => wrapWithTheme(<LibraryRow title={TITLE} thumbnail={THUMBNAIL} type={ItemType.COLLECTION} onClick={() => clickedRow = true }/>);

// snapshots
test("Library row (snapshot)", () => {
    const lr = create(createLibraryRow());
    const serialized = lr.toJSON();
    expect(serialized).toMatchSnapshot();
});
test("Library row fixed width (snapshot)", () => {
    const lr = create(createLibraryRowFixedWidth());
    const serialized = lr.toJSON();
    expect(serialized).toMatchSnapshot();
});
test("Library row with actions (snapshot)", () => {
    const lr = create(createLibraryRowWithActions());
    const serialized = lr.toJSON();
    expect(serialized).toMatchSnapshot();
});
test("Library folder row (snapshot)", () => {
    const lr = create(createLibraryFolderRow());
    const serialized = lr.toJSON();
    expect(serialized).toMatchSnapshot();
});

// enzyme
test("Library row (mount: enzyme+jsdom)", () => {
    const fmMount = mount(createLibraryRow());
    const thumbnail = fmMount.find(".thumbnail").first();
    const title = fmMount.find(".library-row-title-label").first().text();
    expect(thumbnail.props().src).toEqual(THUMBNAIL);
    expect(title).toEqual(TITLE);
});

test("Library row fixed width (mount: enzyme+jsdom)", () => {
    const fmMount = mount(createLibraryRowFixedWidth());
    const row = fmMount.find(".library-row").first();
    expect(row.props().style.width).toEqual(WIDTH);
});

test("Library row with actions (mount: enzyme+jsdom)", () => {
    const fmMount = mount(createLibraryRowWithActions());
    const contextMenuIcon = fmMount.find(".context-menu").first();
    expect(contextMenuIcon).toHaveLength(1);
});
