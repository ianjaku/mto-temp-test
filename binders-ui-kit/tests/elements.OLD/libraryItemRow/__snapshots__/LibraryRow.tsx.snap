// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library folder row (snapshot) 1`] = `
<div
  className="library-row"
  style={Object {}}
>
  <div
    className="library-row-image"
    onClick={[Function]}
  >
    <div
      className="thumbnail-outer-wrapper"
      style={
        Object {
          "backgroundColor": "#white",
          "height": 86,
          "width": 86,
        }
      }
    >
      <div
        className="thumbnail-wrapper fit landscape"
        onClick={undefined}
        onDoubleClick={undefined}
        onMouseDownCapture={undefined}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "backgroundColor": "#white",
            "height": 86,
            "width": 86,
          }
        }
      >
        <img
          className="thumbnail fit landscape"
          onLoad={[Function]}
          src="https://dummyimage.com/100x100"
          style={
            Object {
              "height": "auto",
              "width": 86,
            }
          }
          title="https://dummyimage.com/100x100"
        />
      </div>
    </div>
  </div>
  <div
    className="library-row-title"
    onClick={[Function]}
  >
    <label
      className="library-row-title-label"
    >
      a document
    </label>
    <span
      className="material-icons library-row-title-icon"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontSize": 24,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
    >
      folder_open
    </span>
  </div>
</div>
`;

exports[`Library row (snapshot) 1`] = `
<div
  className="library-row"
  style={Object {}}
>
  <div
    className="library-row-image"
    onClick={[Function]}
  >
    <div
      className="thumbnail-outer-wrapper"
      style={
        Object {
          "backgroundColor": "#white",
          "height": 86,
          "width": 86,
        }
      }
    >
      <div
        className="thumbnail-wrapper fit landscape"
        onClick={undefined}
        onDoubleClick={undefined}
        onMouseDownCapture={undefined}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "backgroundColor": "#white",
            "height": 86,
            "width": 86,
          }
        }
      >
        <img
          className="thumbnail fit landscape"
          onLoad={[Function]}
          src="https://dummyimage.com/100x100"
          style={
            Object {
              "height": "auto",
              "width": 86,
            }
          }
          title="https://dummyimage.com/100x100"
        />
      </div>
    </div>
  </div>
  <div
    className="library-row-title"
    onClick={[Function]}
  >
    <label
      className="library-row-title-label"
    >
      a document
    </label>
    
  </div>
</div>
`;

exports[`Library row fixed width (snapshot) 1`] = `
<div
  className="library-row"
  style={
    Object {
      "width": "600px",
    }
  }
>
  <div
    className="library-row-image"
    onClick={[Function]}
  >
    <div
      className="thumbnail-outer-wrapper"
      style={
        Object {
          "backgroundColor": "#white",
          "height": 86,
          "width": 86,
        }
      }
    >
      <div
        className="thumbnail-wrapper fit landscape"
        onClick={undefined}
        onDoubleClick={undefined}
        onMouseDownCapture={undefined}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "backgroundColor": "#white",
            "height": 86,
            "width": 86,
          }
        }
      >
        <img
          className="thumbnail fit landscape"
          onLoad={[Function]}
          src="https://dummyimage.com/100x100"
          style={
            Object {
              "height": "auto",
              "width": 86,
            }
          }
          title="https://dummyimage.com/100x100"
        />
      </div>
    </div>
  </div>
  <div
    className="library-row-title"
    onClick={[Function]}
  >
    <label
      className="library-row-title-label"
    >
      a document
    </label>
    
  </div>
</div>
`;

exports[`Library row with actions (snapshot) 1`] = `
<div
  className="library-row"
  style={Object {}}
>
  <div
    className="library-row-image"
    onClick={[Function]}
  >
    <div
      className="thumbnail-outer-wrapper"
      style={
        Object {
          "backgroundColor": "#white",
          "height": 86,
          "width": 86,
        }
      }
    >
      <div
        className="thumbnail-wrapper fit landscape"
        onClick={undefined}
        onDoubleClick={undefined}
        onMouseDownCapture={undefined}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "backgroundColor": "#white",
            "height": 86,
            "width": 86,
          }
        }
      >
        <img
          className="thumbnail fit landscape"
          onLoad={[Function]}
          src="https://dummyimage.com/100x100"
          style={
            Object {
              "height": "auto",
              "width": 86,
            }
          }
          title="https://dummyimage.com/100x100"
        />
      </div>
    </div>
  </div>
  <div
    className="library-row-title"
    onClick={[Function]}
  >
    <label
      className="library-row-title-label"
    >
      a document
    </label>
    
    <div
      className="context-menu"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "display": "inline-block",
          "muiPrepared": true,
          "position": "relative",
        }
      }
    >
      <button
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            className="contextMenu-icon material-icons"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fontSize": 24,
                "muiPrepared": true,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
              }
            }
          >
            more_vert
          </span>
        </div>
      </button>
      <div
        style={
          Object {
            "display": "none",
          }
        }
      />
    </div>
  </div>
</div>
`;
